### ISSUE TITLE
Subplots of 3D traces  with common coloraxis are incorrectly colormapped

### ISSUE THREAD

**Issue by empet** [plotly/plotly.py#2130]
Example:

```
x= np.linspace(-1, 1, 100)
x,  y = np.meshgrid(x,x)
z1 = 2*np.sin(x**2*y)

scatt = go.Scatter3d(x=[2, 5, 1, 6], y= [-3, 0, -1, 3], z = [4, 2, 6, 3],
                     mode='markers', marker_coloraxis='coloraxis', 
                     marker_size=10, marker_color=[2, 5,1, 4])

surf1 = go.Surface(x=x, y=y, z=z1, coloraxis = 'coloraxis')

fig1 = make_subplots(
          rows=1, cols=2,
          subplot_titles=(''),
          horizontal_spacing=0.1,
          specs=[[{"type": "scene"}]*2]) 


fig1.add_trace(surf1, 1, 1)
fig1.add_trace(scatt, 1, 2)

fig1.update_layout(width=800, height=600,
                   scene2_xaxis_range=[0.5, 6.5], scene2_zaxis_range=[1.5, 6.5],
                   coloraxis = dict(colorscale='balance', colorbar_len=0.7))
fig1.update_scenes(camera_eye_x=1.75, camera_eye_y=1.75, camera_eye_z=1.75);
```
fig1:

![surf-scatt](https://user-images.githubusercontent.com/3627253/73161375-0f3bda00-40f4-11ea-9a20-a5d9d3b15528.png)

Plotting the surface in an independent figure we get:
![figsurf](https://user-images.githubusercontent.com/3627253/73161453-35fa1080-40f4-11ea-85c4-46e83581abb7.png)

More examples in this notebook [https://plot.ly/~empet/15465](https://plot.ly/~empet/15465)

**Comment by emmanuelle** [comment]
Hi @empet , I don't understand why you say the colormapping is incorrect : the purpose of a common coloraxis is to have a shared colorscale for the two subplots (or the two traces, they could also be in the same subplot). Therefore the bounds which are used for the colorscale are computed using the data of all traces. If you don't want this behaviour you can use a separate colorscale for each trace. Please tell me if I'm missing something, and which behaviour you would have expected.

**Comment by emmanuelle** [comment]
I also took a look at the notebook, and you would have the expected behaviour using a separate colorscale for each subplot.

**Comment by empet** [comment]
Hi @emmanuelle ,
You are right!
Well, I started after many attempts to understand a real issue (I think) related to Mesh3d with cell intensity. Being confused of what is displayed, 
I defined  a subplot of one row and two columns with the two Mesh3d traces in the second plot. 
I tried to define for each one a  hovertemplate that leads to displaying the vertex, respectively cell intensities. For the latter case by default Plotly anchors the hoverbox at the points defined in the trace definition (in this case the vertices), but intensities are associated to faces. Hence displaying a vertex coordinates and a cell intensity can be confusing, because that vertex can be common to many faces. Then I defined as customdata the coordinates of face centers, but obviously the hoverbox remained anchored at vertices.

Besides this drawback  I noticed that the Mesh3d with `intensitymode='cell'` was colored with an almost constant color, using the same `coloraxis`. And I said *"bug in subplots of 3d traces"*. I left aside the hovertemplate issue, and without taking into  account that the two meshes are not colored according to the z-values (that are common for the two meshes), but according to their intensities, I concluded: colormapping issue.
    
    Then without looking at the range of colormapped values  "I checked if the same issue" manifests for other combinations of 3d traces. So I defined the first subplot and the second one. I looked only at their colors, not   at the range of colormapped value to realize that it's all OK. 
Thanks for noticing  that I was wrong!!!!
    
I'll open an issue on plotly.js to suggest finding a method to display the cell intensities in a relevant way. For geometry processing   it would be very important to anchor the hoverbox at the center of mass of a hovered face(triangle) and display it's intensity
    that can be the face area, the length of the face normal, and more. Please look at this plot  [https://plot.ly/~empet/15474](https://plot.ly/~empet/15474)
    to understand what I'm referring to  and think about a method to 
    remove this drawback.
